README.md

```diff
@@ -1,6 +1,261 @@
-# Flexible Streamer for SNAX
+# Flexible Streamer Generator for SNAX
+
+The Flexible Streamer Generator is built to alleviate the data layout concern of an accelerator. It decouples the the accelerator with the real memory system. It simplifies the data interface of the accelerator. The accelerator doesn't need to consider the complicated request and response to/from the real memory system. The accelerator's data interface is simply a valid-ready decoupled handshake interface. Besides, the pre-fetch mechanism inside the Streamer also helps to relieve the data contention problem.
```

Comment:
> Why not name it as a SNAX streamer? haha for branding purposes?

* It is used for SNAX but it can be used in general cases too. I found ETH also has its `hwpe-stream` and UCB has its `memstreamer`. But actually, our streamer can cover their functionality to some extent. I am thinking about extending this streamer to a more general one later... (in the long term, maybe).
* I think `Flexible Streamer Generator for SNAX`  is a little bit long!